optuna/storages/_journal/redis.py

```diff
@@ -82,5 +85,30 @@ def append_logs(self, logs: List[Dict[str, Any]]) -> None:
                 log_number = self._redis.incr(f"{self._prefix}:log_number", 1)
                 self._redis.set(self._key_log_id(log_number), json.dumps(log))
 
+    def save_snapshot(self, snapshot: bytes) -> None:
+        self._redis.setnx(f"{self._prefix}:snapshot_version", -1)
+        snapshot_version = self._redis.incr(f"{self._prefix}:snapshot_version", 1)
```

Comment:
Do we actually need the snapshot version?
I'm concerned that saving each snapshot using a separate key could consume very large storage space if the number of trials / studies is too many.

IMO, using the same key for snapshots and simply overwriting the one if it exists could work.

Reply:
Thanks for the comment. It is reasonable. I removed the snapshot_version. The load_snapshot does not retry even if the snapshot is not written.

Reply:
Thanks!